src/main/core/sim_stats.rs

```diff
@@ -0,0 +1,27 @@
+use crate::utility::counter::Counter;
+
+use serde::Serialize;
+
+#[derive(Serialize, Clone, Debug)]
+pub struct ObjectStats {
+    pub alloc_counts: Counter,
+    pub dealloc_counts: Counter,
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct SimStats {
+    pub objects: ObjectStats,
+    pub syscalls: Counter,
+}
+
+impl SimStats {
+    pub fn new() -> Self {
+        Self {
+            objects: ObjectStats {
+                alloc_counts: Counter::new(),
+                dealloc_counts: Counter::new(),
+            },
+            syscalls: Counter::new(),
+        }
+    }
+}
```

Comment:
Should this end in a newline?

Reply:
There should be a newline there, github would put a red icon there if there was no newline at the end of the file.